### PR TITLE
#113: fix `ERROR: not found` CI failures

### LINKED ISSUE #113: fix `ERROR: not found` CI failures
The error, as seen [here](https://github.com/Pressio/rom-tools-and-workflows/actions/runs/7693113245/job/20961234702):
```
 ============================ no tests ran in 0.25s =============================
ERROR: not found: /home/runner/work/rom-tools-and-workflows/rom-tools-and-workflows/tests/romtools/__pycache__
(no match in any of [<Dir romtools>])
```

First appeared on [this commit](https://github.com/Pressio/rom-tools-and-workflows/commit/5f8e5e8a415852e79c0667c1d5e80d5f40875b52).

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #113

Also deletes an extraneous `docs.yaml` pipeline that seems to have been superceded by `deploy_docs.yaml`.

### CODE REVIEW

File: .github/workflows/install-and-test.yaml [hunk 1]
```diff
@@ -76,24 +78,24 @@ jobs:
...
+      # - name: Run tests (MPI, n=3)
+      #   run: |
+      #     mpirun -n 3 python -m pytest tests/romtools/* --with-mpi --timeout=300
 
-      - name: Run tests (MPI, n=4)
-        run: |
-          mpirun -n 4 python -m pytest tests/romtools/* --with-mpi --timeout=300
+      # - name: Run tests (MPI, n=4)
+      #   run: |
+      #     mpirun -n 4 python -m pytest tests/romtools/* --with-mpi --timeout=300
```
Comment (fnrizzi) [inline .github/workflows/install-and-test.yaml]: why are tehse all commented out?

Comment (fnrizzi) [reply .github/workflows/install-and-test.yaml]: why are tehse all commented out?

**Review by fnrizzi — CHANGES_REQUESTED** [review]

File: .github/workflows/install-and-test.yaml (hunk 1 above)
Comment (cwschilly) [reply .github/workflows/install-and-test.yaml]: Sorry--leftover from debugging. Just fixed them

**Comment by fnrizzi** [comment]
is this ready?

**Comment by fnrizzi** [comment]
@cwschilly can you please add a comment to say why we are pinning the pytest version?

**Comment by cwschilly** [comment]
@fnrizzi This is ready now

**Review by fnrizzi — APPROVED** [review]

**Pull request merged**